packages/editor/src/components/editor-container/editor-container.ts

```diff
@@ -3,13 +3,36 @@ import { customElement, property, query, state } from 'lit/decorators.js';
 import { choose } from 'lit/directives/choose.js';
 
 import type { Space } from '@blocksuite/store';
+import { DisposableGroup, Signal } from '@blocksuite/store';
 import { ClipboardManager, ContentParser } from '../..';
 import { BlockSchema } from '../../block-loader';
 
 type PageBlockModel = InstanceType<typeof BlockSchema['affine:page']>;
 
+declare global {
+  interface HTMLElementTagNameMap {
+    'editor-container': EditorContainer;
+  }
+}
+
 @customElement('editor-container')
 export class EditorContainer extends LitElement {
+  /**
+   * @internal
+   * Resolve the readiness of the editor, on {@link EditorContainer.editorReady}
+   *
+   * TODO: rootAdded signal doesn't seem to trigger for completely empty documents...
+   * So, in cases where we stat on a blank page, we won't know the editor is ready...
+   */
+  #resolveReady: undefined | (() => void) = undefined;
```

Comment:
Just a question, why not use `private` from TypeScript?

Reply:
I want to indicate that this is absolutely not supposed to be exposed to the outside. So, I want to prevent monkey patching.

I don't feel strongly, though.

Reply:
I think we can just stick to the `private resolveReady` convention, this aligns well with different contributors.

Reply:
I will apply that suggestion